cmd/keeper/cmd/keeper.go

```diff
@@ -1812,25 +1826,35 @@ func keeper(c *cobra.Command, args []string) {
 		die("cannot create data dir: %v", err)
 	}
 
-	if cfg.pgListenAddress == "" {
-		die("--pg-listen-address is required")
+	if cfg.pgAdvertiseAddress == "" && cfg.pgListenAddress == "" {
+		die("--pg-listen-address and/or --pg-advertise-address is required")
+	} else if cfg.pgListenAddress == "" {
+		cfg.pgListenAddress = "*"
+	} else if cfg.pgAdvertiseAddress == "" {
+		cfg.pgAdvertiseAddress = cfg.pgListenAddress
 	}
 
-	ip := net.ParseIP(cfg.pgListenAddress)
+	if cfg.pgAdvertisePort == "" && cfg.pgPort == "" {
+		die("--pg-port and/or --pg-advertise-port is required")
+	} else if cfg.pgPort == "" {
+		cfg.pgPort = cfg.pgAdvertisePort
+	} else if cfg.pgAdvertisePort == "" {
+		cfg.pgAdvertisePort = cfg.pgPort
+	}
+
+	ip := net.ParseIP(cfg.pgAdvertiseAddress)
 	if ip == nil {
-		log.Warnf("provided --pgListenAddress %q: is not an ip address but a hostname. This will be advertized to the other components and may have undefined behaviors if resolved differently by other hosts", cfg.pgListenAddress)
+		log.Warnf("provided --pg-advertise-address %q: is not an ip address but a hostname. This will be advertized to the other components and may have undefined behaviors if resolved differently by other hosts", cfg.pgAdvertiseAddress)
```

Comment:
The log can look strange to an user if it hasn't provided pgAdvertiseAddress. Perhaps to could be change to 
`PGAdvertiseAddress: %q: ...` and when setting pgAdvertiseAddress to pgListenAddress logging that we are setting PGAdvertiseAddress to PGListenAddress.

Reply:
Yes as following previous comment, I think this must be change to apply correct message.